src/java/org/apache/cassandra/cql3/statements/DeleteStatement.java

```diff
@@ -174,7 +186,8 @@ protected ModificationStatement prepareInternal(ClientState state,
                                                        operations,
                                                        restrictions,
                                                        conditions,
-                                                       attrs);
+                                                       attrs,
+                                                       bounds.describeSource(enclosingStatement));
```

Comment:
this feels backwards, we create the underline statements before we create the composite statement, so how do we pass it in?

Reply:
No, notice that the enclosing composite statement is a parsed statement and it already exists so as the parsed delete statement.